packages/template-retail-react-app/app/hooks/use-derived-product.js

```diff
@@ -38,7 +39,35 @@ export const useDerivedProduct = (
         isProductPartOfBundle
     )
     const [quantity, setQuantity] = useState(initialQuantity)
+    const {site} = useMultiSite()
 
+    // Helper function to get inventory by ID
+    const getInventoryById = (inventoryId) => {
```

Comment:
This seems like we can put this in a util file or put it outside of the component. Putting it inside the component here will cause this function to reinitialize every time it renders.

Reply:
```
const getInventoryById = (inventoryId, product) => {}
```